core/src/main/java/com/graphhopper/isochrone/algorithm/ShortestPathTree.java

```diff
@@ -158,13 +164,22 @@ public void search(int from, final Consumer<IsoLabel> consumer) {
                     fromMap.put(nextTraversalId, label);
                     if (getExploreValue(label) <= limit) {
                         queueByWeighting.add(label);
+                    } else if (includeOverextendedEdges) {
+                        // Overextended case
+                        double overextension = limit - getExploreValue(currentLabel);
+                        label.distance_consumed = OptionalDouble.of(overextension);
+                        consumer.accept(label);
```

Comment:
It looks like they already have a function for doing linear interpolation (https://github.com/graphhopper/graphhopper/blob/069c6510bdf9325490e70f87ded920f81f5fd7cf/core/src/main/java/com/graphhopper/reader/dem/EdgeSampling.java#LL57C55-L57C55), but looking into EdgeIterator it looks like it should be possible to return the actual road geometry of an edge (https://github.com/graphhopper/graphhopper/blob/069c6510bdf9325490e70f87ded920f81f5fd7cf/core/src/main/java/com/graphhopper/util/EdgeIteratorState.java#L120) which would be extremely helpful to include in the response (as an option, as it would increase the size of the response but might have a negligible impact on the response time). If we have the way geometry, I wonder if there's an easier/faster way to interpolate that for the edges that cross over the limit?

Reply:
Hmm we don't have to interpolate anything, as we just have to check how much of the edge is "over" the limit. 

As for returning the actual road geometry, I'd prefer to do that in a second PR to graphhopper. I'm not sure what their standards are about PR's, but getting this simple fix in means an immediate boost to transify. But returning more accurate road geometries would just be a nice quality-of-life improvement.

What do you think?

Reply:
Yes - agree with keeping the scope of this PR as it is. For this PR, could you put it into a JAR so I can test it/deploy it? And yes - you can make a PR for merging this into graphhopper for them to review.

For returning the road geometry I think this could be really helpful to visualize in Transify (I have a previous failed attempt in using their snapping endpoint to do this as it was super slow and sometimes crashed) + make it easy to interpolate the edge in GH (so no more post-processing in python required), but agree it should be in another PR - is that something you'd be good with working on next?

Reply:
Yes for both. Let's see if this works and is a good approach, and I'll start working on returning better road edges. 

Sending the JAR on Teams. **Note that since I haven't gotten the transify-routing repo setup yet, I haven't tested the changes to the API yet.**

Reply:
Sounds good! The changes look good to me.